File: python/find-winner-tic-tac-toe.py
class Solution:
    def tictactoe(self, moves: List[List[int]]) -> str:
        game_board = [[""]*3 for _ in range(3)]
        Alex = True
        for r, c in moves:
            game_board[r][c] = "X" if Alex else "O"
            Alex = Alex ^ True
        
        def checkRows(game_board, char):
            for row in game_board:
                if "".join(row) == char*3:
                    return True
            return False
        def checkCols(game_board, char):
            for col in list(zip(*game_board)):
                print(col)
                if "".join(col) == char*3:
                    return True
            return False
        def checkDiag(game_board, char):
            return (
                "".join([game_board[i][i] for i in range(3)]) == char*3 or
                "".join([game_board[i][2-i] for i in range(3)]) == char*3
            )
        
        if checkRows(game_board, "X") or checkCols(game_board, "X") or checkDiag(game_board, "X"): return "A"
        if checkRows(game_board, "O") or checkCols(game_board, "O")or checkDiag(game_board, "O"): return "B"
        
        for row in game_board:
            for cell in row:
                if cell == "":
                    return "Pending"
        
        return "Draw"

File: go/find_winner_tic_tac_toe.go
package leeetcode

func tictactoe(moves [][]int) string {
	check_rows := func(game_board [3][3]string, char string) bool {
		for _, row := range game_board {
			if row[0] == char && row[1] == char && row[2] == char {
				return true
			}
		}
		return false
	}

	check_cols := func(game_board [3][3]string, char string) bool {
		for i := 0; i < 3; i++ {
			if game_board[0][i] == char && game_board[1][i] == char && game_board[2][i] == char {
				return true
			}
		}
		return false
	}

	check_diags := func(game_board [3][3]string, char string) bool {
		if game_board[0][0] == char && game_board[1][1] == char && game_board[2][2] == char {
			return true
		} else if game_board[0][2] == char && game_board[1][1] == char && game_board[2][0] == char {
			return true
		}
		return false
	}

	var game_board [3][3]string
	alex := true

	for _, el := range moves {
		if alex == true {
			game_board[el[0]][el[1]] = "X"
		} else {
			game_board[el[0]][el[1]] = "O"
		}
		alex = !alex
	}

	if check_rows(game_board, "X") || check_cols(game_board, "X") || check_diags(game_board, "X") {
		return "A"
	}
	if check_rows(game_board, "O") || check_cols(game_board, "O") || check_diags(game_board, "O") {
		return "B"
	}

	for _, row := range game_board {
		for _, cell := range row {
			if cell == "" {
				return "Pending"
			}
		}
	}

	return "Draw"
}
